src/backend/gl/src/lib.rs

```diff
@@ -656,14 +656,32 @@ impl CommandQueue {
             },
             Command::CopyBuffer(src, dst, src_offset, dst_offset, size) => {
                 let gl = &self.share.context;
-                unsafe {
-                    gl.BindBuffer(gl::COPY_READ_BUFFER, src);
-                    gl.BindBuffer(gl::COPY_WRITE_BUFFER, dst);
-                    gl.CopyBufferSubData(gl::COPY_READ_BUFFER,
-                                         gl::COPY_WRITE_BUFFER,
-                                         src_offset,
-                                         dst_offset,
-                                         size);
+
+                if self.share.capabilities.copy_buffer_supported {
+                    unsafe {
+                        gl.BindBuffer(gl::COPY_READ_BUFFER, src);
+                        gl.BindBuffer(gl::COPY_WRITE_BUFFER, dst);
+                        gl.CopyBufferSubData(gl::COPY_READ_BUFFER,
+                                            gl::COPY_WRITE_BUFFER,
+                                            src_offset,
+                                            dst_offset,
+                                            size);
+                    }
+                } else {
```

Comment:
Maybe add `debug_assert!(self.share.private_caps.buffer_storage_supported == false)` to make sure there is no persistent mapping going on.